packages/plugin/src/node/hosted-plugin-manager.ts

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable, named } from "inversify";
+import * as cp from "child_process";
+import * as fs from "fs";
+import * as net from "net";
+import URI from '@theia/core/lib/common/uri';
+import { ContributionProvider } from "@theia/core/lib/common/contribution-provider";
+import { HostedPluginUriPostProcessor } from "./hosted-plugin-uri-postprocessor";
+
+export const HostedPluginRunner = Symbol('HostedPluginRunner');
+
+/**
+ * Is responsible for running and handling separate Theia instance with given plugin.
+ */
+export interface HostedPluginRunner {
```

Comment:
I think the name is misleading. It's running a theia instance + a plugin
WDYT on `HostedTheiaRunner` with a `runPlugin()` method

Reply:
Yes, it should be `HostedTheiaRunner`, thanks